src/main/java/omero/gateway/ServerInformation.java

```diff
@@ -30,57 +34,134 @@
 
 public class ServerInformation {
 
-    /** The hostname */
-    private String hostname;
-
-    /** The port */
-    private int port;
+    /** The URI */
+    private URI uri;
```

Comment:
If possible, I'd make all of these fields `final` to keep the equals and hashCode implementations stable.

Reply:
That's really tricky, as the URI class doesn't have setters. Ideally this should be an URL anyway, but URL doesn't support `wss` etc.

Reply:
👍 Simply removed equals/hashcode implementation.